lib/statsd.rb

```diff
@@ -21,6 +21,7 @@ def initialize(opts={})
       @host = opts[:host] || 'localhost'
       @port = opts[:port] || 8125
       @prefix = opts[:prefix]
+      @socket = UDPSocket.new
```

Comment:
I'm thinking we need to thread protect this. I'm not sure what happens to the UDPSocket that is garbage collected without being closed.